src/webextension/background/message-ports.js

```diff
@@ -23,6 +26,28 @@ export default function initializeMessagePorts() {
 
   browser.runtime.onMessage.addListener(async function(message, sender) {
     switch (message.type) {
+    case "open_view":
+      return openView(message.name);
+    case "close_view":
+      return closeView(message.name);
+
+    case "signin":
+      return authz.signIn(message.interactive);
+    case "initialize":
+      return authz.verify(message.email, message.password).
+        then(() => openDataStore()).
+        then((ds) => ds.initialize({
+          password: message.password,
+        })).
+        then(updateBrowserAction);
```

Comment:
I'm not sure what you mean; is it the following?

```
let ds = await openDataStore();
await ds.initialize(...);
await updateBrowserAction(ds);
```

Reply:
Yeah, although we might want to leave the `.then` for `openDataStore`, since we'd have to create a new scope in the `switch` otherwise. Something like:

```js
openDataStore().then(async(ds) => {
  await ds.initialize(...);
  await updateBrowserAction(ds);
});
```

Or we could just put some braces around this code (or declare `ds` outside of the switch).